### PR TITLE
003: User Login (Backend)

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by DaleCS — CHANGES_REQUESTED** [review]
I think you need this line in server.js below the other routes:
app.use("/api/login", require("./routes/api/login"));

I get this error when I try to run the backend server:
Capture

Other than those two, the rest of the code looks good and straight forward.

**Review by DaleCS — CHANGES_REQUESTED** [review]
You need this line at the bottom of your routes file (login.js) otherwise the application won't run.
'module.exports = router;'

I can't seem to get a proper http response from the route as well 'http://localhost:5000/api/login/'. How did you test this?

**Comment by antoinengu** [comment]
Everything should work now. Swapped over to using passport, which will make authenticating routes easier.

send POST to: http://localhost:5000/api/users

In the headers section, make sure that the Content-Type is set to application/json

For the body, raw should be selected, and JSON should be selected from the drop down.

example request
{
"username":"some.email@email.com",
"password":"passsW0rd$"
}

File: config/passport.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const db = require("./db");
+
+passport.use(new LocalStrategy(
+    (username, password, done) => {
+    db.query('SELECT * FROM users WHERE email = ?', username, (err, rows) => {
+        if (err) {
+            return done(err)
+        }
+        const userAccount = rows[0];
+        const passwordMatch  = bcrypt.compare(password, userAccount.password);
```
Comment (DaleCS) [inline config/passport.js]: Need to check if the db query returned nothing. Server crashes when an attempting to access 'userAccount.password'

**Review by DaleCS — CHANGES_REQUESTED** [review]
Almost there! One last minor change request. Other than that everything else works great.

File: config/passport.js (hunk 1 above)
Comment (antoinengu) [reply config/passport.js]: fixed

**Review by DaleCS — APPROVED** [review]
Looks good!

**Pull request merged**